src/scripts/glue_jobs/node_snapshots.py

```diff
@@ -25,37 +24,55 @@ def get_dynamic_frame(connection_type, file_format, source_path, glue_context):
             ("snapshotTimestamp",               "bigint",   "snapshot_timestamp",   "bigint"),
-            ("snapshot.id",                     "bigint",   "id",                   "bigint"),
-            ("snapshot.benefactorId",           "bigint",   "benefactor_id",        "bigint"),
-            ("snapshot.projectId",              "bigint",   "project_id",           "bigint"),
-            ("snapshot.parentId",               "bigint",   "parent_id",            "bigint"),
+            ("snapshot.id",                     "string",   "id",                   "bigint"),
+            ("snapshot.benefactorId",           "string",   "benefactor_id",        "bigint"),
+            ("snapshot.projectId",              "string",   "project_id",           "bigint"),
+            ("snapshot.parentId",               "string",   "parent_id",            "bigint"),
             ("snapshot.nodeType",               "string",   "node_type",            "string"),
             ("snapshot.createdOn",              "bigint",   "created_on",           "bigint"),
             ("snapshot.createdByPrincipalId",   "bigint",   "created_by",           "bigint"),
             ("snapshot.modifiedOn",             "bigint",   "modified_on",          "bigint"),
             ("snapshot.modifiedByPrincipalId",  "bigint",   "modified_by",          "bigint"),
             ("snapshot.versionNumber",          "bigint",   "version_number",       "bigint"),
-            ("snapshot.fileHandleId",           "bigint",   "file_handle_id",       "bigint"),
+            ("snapshot.fileHandleId",           "string",   "file_handle_id",       "bigint"),
             ("snapshot.name",                   "string",   "name",                 "string"),
             ("snapshot.isPublic",               "boolean",  "is_public",            "boolean"),
             ("snapshot.isControlled",           "boolean",  "is_controlled",        "boolean"),
             ("snapshot.isRestricted",           "boolean",  "is_restricted",        "boolean"),
+            ("change_date",                     "string",   "change_date",          "string"),
         ],
-        transformation_ctx="mapped_dynamic_frame")
+        transformation_ctx="mapped_dynamic_frame"
+    )
+
     return mapped_dynamic_frame
 
+def strip_syn_prefix(input_string):
+    if input_string is None:
+        return input_string
+
+    return input_string.removeprefix('syn')
 
-# process the access record
 def transform(dynamic_record):
-    date = datetime.utcfromtimestamp(dynamic_record["change_timestamp"] / 1000.0)
+    date = datetime.utcfromtimestamp(dynamic_record["changeTimestamp"] / 1000.0)
+    
+    # Add the partition date from the timestamp of the change
     dynamic_record["change_date"] = date.strftime("%Y-%m-%d")
+    
+    # The records come in with the syn prefix, we need to remove that
+    dynamic_record["snapshot"]["id"] = strip_syn_prefix(dynamic_record["snapshot"]["id"])
```

Comment:
Just out of curiosity, why do syn ids need to be removed?

Reply:
The idea is that it's much better to have a bigint type in the queriable table (simpler to write queries, joins etc)

Reply:
Makes sense from a compute perspective, maybe a little more confusing from a user perspective (depending on who the audience is), but not too difficult to remove "syn"